src/tests/unit/auto/get_device_list.cpp

```diff
@@ -79,40 +88,72 @@ class GetDeviceListTest : public ::testing::TestWithParam<ConfigParams> {
-};
-
-
-INSTANTIATE_TEST_SUITE_P(smoke_Auto_BehaviorTests, GetDeviceListTest,
-                ::testing::ValuesIn(testConfigs),
-            GetDeviceListTest::getTestCaseName);
+const std::vector<Params> testConfigsWithId = {Params{" ", " "},
+                                         Params{"", "CPU,GPU.0,GPU.1"},
+                                         Params{"CPU, ", "CPU, "},
+                                         Params{" ,CPU", " ,CPU"},
+                                         Params{"CPU,", "CPU"},
+                                         Params{"CPU,,GPU", "CPU,GPU.0,GPU.1"},
+                                         Params{"CPU, ,GPU", "CPU, ,GPU.0,GPU.1"},
+                                         Params{"CPU,GPU,GPU.1", "CPU,GPU.0,GPU.1"},
+                                         Params{"CPU,GPU,VPUX,INVALID_DEVICE", "CPU,GPU.0,GPU.1,VPUX,INVALID_DEVICE"},
+                                         Params{"VPUX,GPU,CPU,-GPU.0", "VPUX,GPU.1,CPU"},
+                                         Params{"-GPU.0,GPU,CPU", "GPU.1,CPU"},
+                                         Params{"-GPU.0,GPU", "GPU.1"},
+                                         Params{"-GPU,GPU.0", "GPU.0"},
+                                         Params{"-GPU.0", "CPU,GPU.1"},
+                                         Params{"-GPU.0,-GPU.1", "CPU"},
+                                         Params{"-GPU.0,-GPU.1,INVALID_DEVICE", "INVALID_DEVICE"},
+                                         Params{"-GPU.0,-GPU.1,-INVALID_DEVICE", "CPU"},
+                                         Params{"-GPU.0,-CPU", "GPU.1"}};
+
+const std::vector<Params> testConfigs = {Params{" ", " "},
+                                         Params{"", "CPU,GPU"},
+                                         Params{"GPU", "GPU"},
+                                         Params{"GPU.0", "GPU.0"},
+                                         Params{"GPU,GPU.0", "GPU"},
+                                         Params{"CPU", "CPU"},
+                                         Params{" ,CPU", " ,CPU"},
+                                         Params{" ,GPU", " ,GPU"},
+                                         Params{"GPU, ", "GPU, "},
+                                         Params{"CPU,GPU", "CPU,GPU"},
+                                         Params{"CPU,-GPU", "CPU"},
+                                         Params{"CPU,-GPU,GPU.0", "CPU,GPU.0"},
+                                         Params{"CPU,-GPU,GPU.1", "CPU,GPU.1"},
+                                         Params{"CPU,GPU,-GPU.0", "CPU"},
+                                         Params{"CPU,GPU,-GPU.1", "CPU,GPU"},
+                                         Params{"CPU,GPU.0,GPU", "CPU,GPU"},
+                                         Params{"CPU,GPU,GPU.0", "CPU,GPU"},
+                                         Params{"CPU,GPU,GPU.1", "CPU,GPU,GPU.1"},
+                                         Params{"CPU,GPU.1,GPU", "CPU,GPU.1,GPU"},
```

Comment:
When there is `GPU.1` in device list, I assume `GPU` is class name. 

I guess the result should be `CPU,GPU.1,GPU.0`

Reply:
The available device list of the platform in this case is {"CPU", "GPU"} rather than {"CPU","GPU.0","GPU.1"}. "GPU.0" should not be existed in the candidate device list.